### PR TITLE
Add Wither & Blood Door Highlight

### PR DESCRIPTION
**Pull request by kevinthegreat1**
Add Wither & Blood Door Highlight!  

Known bugs:  
Some key pickups by other players are not detected. (When the player entity is not loaded or when the key is obtained in a room not yet visited by the client's player.)  
Wither door in the fairy room always remains locked since no key can ever be picked up in the fairy room.

### CODE REVIEW

File: src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/Room.java [hunk 1]
```diff
@@ -43,10 +48,14 @@
 public class Room {
     private static final Pattern SECRET_INDEX = Pattern.compile("^(\\d+)");
     private static final Pattern SECRETS = Pattern.compile("§7(\\d{1,2})/(\\d{1,2}) Secrets");
+    private static final Vec3d DOOR_SIZE = new Vec3d(3, 4, 3);
+    private static final float[] RED_COLOR_COMPONENTS = DyeColor.RED.getColorComponents();
+    private static final float[] GREEN_COLOR_COMPONENTS = DyeColor.GREEN.getColorComponents();
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/Room.java:53]: These should just be plain 1, 0, 0 and 0, 1, 0 like BetterMap, having it be slightly faded is also weird to see.

File: src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/DungeonSecrets.java [hunk 2]
```diff
@@ -76,6 +79,7 @@ public class DungeonSecrets {
     protected static final Logger LOGGER = LoggerFactory.getLogger(DungeonSecrets.class);
     private static final String DUNGEONS_PATH = "dungeons";
     private static final Path CUSTOM_WAYPOINTS_DIR = SkyblockerMod.CONFIG_DIR.resolve("custom_secret_waypoints.json");
+    private static final Pattern KEY_FOUND = Pattern.compile("^(?<name>\\w+) has obtained (?<type>Blood|Wither) Key!$");
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/DungeonSecrets.java:82]: This regex isn't correct as it doesn't account for ranked players:
[MVP+] AzureAaron has obtained Wither Key!

File: src/main/java/de/hysky/skyblocker/config/categories/DungeonsCategory.java [hunk 3]
```diff
@@ -147,6 +147,9 @@ public static ConfigCategory create(SkyblockerConfig defaults, SkyblockerConfig
 								.build())
 						.build())
 
+				//Dungeon Door Highlight
+
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/config/categories/DungeonsCategory.java:151]: This needs config (maybe you forgot)

**Review by AzureAaron — CHANGES_REQUESTED** [review]
Aside from the bugs stated this is it, otherwise its fine to me

File: src/main/java/de/hysky/skyblocker/config/categories/DungeonsCategory.java (hunk 3 above)
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/config/categories/DungeonsCategory.java:170]: Oops I forgot lol.

**Review by AzureAaron — APPROVED** [review]

**Comment by pigerstreet** [comment]
zzz

**Comment by pigerstreet** [comment]
Seems to work fine, however sometimes when I get the key the highlight stays red.

**Pull request merged**